### PR TITLE
Fix compile warnings

### PR DESCRIPTION
**Pull request by maslowalex**
Rebased with upstream repo, fixed minor compile issues.

### CODE REVIEW

File: lib/quark/fixed_point.ex [hunk 1]
```diff
@@ -77,9 +77,11 @@ defmodule Quark.FixedPoint do
   defpartial turing(fun), do: turing_inner().(turing_inner()).(fun)
 
   defpartialp turing_inner(x, y) do
-    cx = curry(x)
-    cy = curry(y)
-    cy.(&(cx.(cx).(cy).(&1)))
+    if turing_inner(x, y) === turing_inner(x, y) do
```
Comment (cognivore) [inline lib/quark/fixed_point.ex]: I don't like calling functions to silence a warning. What was the warning that you're fixing like this?

File: lib/quark/ski.ex [hunk 2]
```diff
@@ -52,7 +52,7 @@ defmodule Quark.SKI do
 
   """
   @spec k(any, any) :: any
-  defpartial k(x, y), do: x
+  defpartial k(x, y), do: if y === y, do: x
```
Comment (cognivore) [inline lib/quark/ski.ex:55]: Doesn't 

```
defpartial k(x, y) do
  x
  y
end
```

work?

File: lib/quark/ski.ex [hunk 3]
```diff
@@ -75,7 +75,7 @@ defmodule Quark.SKI do
 
   """
   @spec second(any, any) :: any
-  defpartial second(_a, b), do: b
+  defpartial second(a, b), do: if a === a, do: b
```
Comment (cognivore) [inline lib/quark/ski.ex:78]: Same as above.

**Review by cognivore — CHANGES_REQUESTED** [review]

File: lib/quark/ski.ex (hunk 2 above)
Comment (maslowalex) [reply lib/quark/ski.ex:55]: Unfortunately, no.

File: lib/quark/fixed_point.ex (hunk 1 above)
Comment (maslowalex) [reply lib/quark/fixed_point.ex]: Fixed

**Comment by cognivore** [comment]
lgtm

**Pull request merged**